### PR TITLE
Support CreatePersonalAccessTokenForCurrentUser endpoint

### PR DESCRIPTION
**Pull request by timofurrer**
Support new endpoint: https://docs.gitlab.com/ee/api/users.html#create-a-personal-access-token-with-limited-scopes-for-the-currently-authenticated-user

### CODE REVIEW

**Comment by timofurrer** [comment]
@svanharmelen this is ready for your review! As usual, I'd be super happy if a merge is followed by an immediate release 😇

File: users.go [hunk 1]
```diff
@@ -1301,6 +1301,38 @@ func (s *UsersService) CreatePersonalAccessToken(user int, opt *CreatePersonalAc
 	return t, resp, nil
 }
 
+// CreatePersonalAccessTokenForCurrentUserOptions represents the available
+// CreatePersonalAccessTokenForCurrentUser() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/users.html#create-a-personal-access-token
```
Comment (svanharmelen) [inline users.go:1308]:
```suggestion
// https://docs.gitlab.com/ee/api/users.html#create-a-personal-access-token-with-limited-scopes-for-the-currently-authenticated-user
```

File: testdata/post_user_personal_access_tokens.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  "revoked": false,
+  "created_at": "2020-10-14T11:58:53.526Z",
+  "scopes": [
+    "k8s_proxy"
+  ],
+  "user_id": 42,
+  "active": true,
+  "expires_at": "2020-10-15",
+  "token": "glpat-aaaaaaaa-bbbbbbbbb"
+}
```
Comment (svanharmelen) [inline testdata/post_user_personal_access_tokens.json:13]: Please end the file with a newline (can be configured in your editor to do this by default)

**Review by svanharmelen — CHANGES_REQUESTED** [review]
If you update the two minor comments, I'll merge it afterwards...

**Comment by timofurrer** [comment]
@svanharmelen done :)

**Review by svanharmelen — APPROVED** [review]
Thanks 👍🏻

**Pull request merged**